hamt/hamt.go

```diff
@@ -427,18 +424,33 @@ func (ds *Shard) ForEachLink(ctx context.Context, f func(*ipld.Link) error) erro
 	})
 }
 
+// EnumLinksAsync returns a channel which will receive Links in the directory
+// as they are enumerated, where order is not gauranteed
+func (ds *Shard) EnumLinksAsync(ctx context.Context) (<-chan format.LinkResult, error) {
```

Comment:
We should add some logic to handle context cancellations and avoid goroutine leaks. (This comment may apply in other parts of the PR.)

Reply:
will look at this